site/src/components/Button.tsx

```diff
@@ -12,17 +79,21 @@ type ButtonProps = {
 
 // probably rename to BPButton?
 // @todo implement loading
-export const Button: FC<ButtonProps> = ({ children, squared, ...props }) => {
-  return (
-    <MuiButton
-      {...props}
-      sx={{
-        borderRadius: squared ? "6px" : undefined,
-        ...props.sx,
-        lineHeight: "1.5",
-      }}
-    >
-      {children}
-    </MuiButton>
-  );
-};
+export const Button: FC<ButtonProps> = forwardRef(
+  ({ children, squared, loading, ...props }, ref) => {
+    return (
+      <MuiButton
+        {...props}
+        sx={{
+          borderRadius: squared ? "6px" : undefined,
+          /** @todo: figure out how to properly merge this object with `props.sx` */
+          ...props.sx,
+          lineHeight: "1.5",
+        }}
+        ref={ref}
```

Comment:
You can do 
```suggestion
  ({ children, squared, loading, sx, ...props }, ref) => {
    return (
      <MuiButton
        {...props}
        sx={{
          borderRadius: squared ? "6px" : undefined,
          ...sx,
          lineHeight: "1.5",
        }}
        ref={ref}
```

Reply:
Unless I'm missing something, that would be equivalent. What I'm referring to here is the fact that the `sx` prop can be a function, and when that is the case cannot be spreaded using `..props.sx` (or `...sx`). For now this means don't provide the a function to the `sx` prop of the custom `Button` wrapper. A better long-term solution might be to enforce this by modifying the type of the `sx` prop for this component, or preferably figuring out a way to properly merge `props.sx` with the extended styling.

Reply:
Ohh... I assumed the issue was the fact that we were assigning the `sx` prop twice. I get what you mean now.